File: python/Glacier2/session/server/poke_session/user_id_resolver.py
# Copyright (c) ZeroC, Inc.

from typing import Protocol, runtime_checkable


@runtime_checkable
class UserIdResolver(Protocol):
    """
    Resolves a user ID from a session token.
    """

    def getUserId(self, token: str) -> str | None:
        """
        Gets the user ID associated with the specified session token, or None if not found.

        Parameters
        ----------
        token : str
            The session token.

        Returns
        -------
        str | None
            The user ID associated with the specified session token, or None if not found.
        """
        ...

    def removeToken(self, token: str) -> None:
        """
        Removes the specified session token.

        Parameters
        ----------
        token : str
            The session token.
        """
        ...

File: swift/Glacier2/Session/Sources/Server/UserIdResolver.swift
// Copyright (c) ZeroC, Inc.

import Foundation
import Glacier2
import Ice

/// Protocol for resolving user IDs from session tokens.
protocol UserIdResolver {
    /// Gets the user ID associated with the specified session token.
    /// - Parameter token: The session token.
    /// - Returns: The user ID associated with the token, or nil if not found.
    func getUserId(_ token: String) -> String?

    /// Removes the specified session token.
    /// - Parameter token: The session token to remove.
    func removeToken(_ token: String)
}
